mne_bids/path.py

```diff
@@ -873,15 +883,25 @@ def get_entities_from_fname(fname):
     idx_key = 0
     for match in re.finditer(param_regex, op.basename(fname)):
         key, value = match.groups()
-        if key not in fname_vals:
-            raise KeyError(f'Unexpected entity "{key}" found in '
-                           f'filename "{fname}"')
-        if fname_vals.index(key) < idx_key:
-            raise ValueError(f'Entities in filename not ordered correctly.'
-                             f' "{key}" should have occurred earlier in the '
-                             f'filename "{fname}"')
-        idx_key = fname_vals.index(key)
-        params[ALLOWED_PATH_ENTITIES_SHORT[key]] = value
+
+        if on_error in ['raise', 'warn']:
```

Comment:
Let's make it a tuple
```suggestion
        if on_error in ('raise', 'warn'):
```